tests/integration/targets/win_certificate_data/tasks/tests.yml

```diff
@@ -0,0 +1,102 @@
+---
+- name: Get stats on a store that doesn't exist
+  ansible.windows.win_certificate_data:
+    store_name: teststore
+  register: test_store
+
+- name: Ensure test store does not exist
+  ansible.builtin.assert:
+    that:
+      - test_store.exists == false
+
+- name: Get stats on the Root certificate store
+  ansible.windows.win_certificate_data:
+    store_name: Root
+  register: root_store
+
+- name: Verify at least one certificate is present in Root store
+  ansible.builtin.assert:
+    that:
+      - root_store.exists
+      - root_store.certificates | length > 0
+
+- name: Get stats on a certificate that does not exist
+  ansible.windows.win_certificate_data:
+    thumbprint: "ABC"
+  register: actual
+
+- name: Ensure non-existent certificate has exists set to false
+  ansible.builtin.assert:
+    that:
+      - actual.exists == false
+
+- name: Get stats on root certificate by thumbprint
+  ansible.windows.win_certificate_data:
+    thumbprint: "{{ root_thumbprint }}"
+    store_name: Root
+  register: root_stats
+
+- name: Validate root certificate stats
+  ansible.builtin.assert:
+    that:
+      - root_stats.exists
+      - root_stats.certificates[0].archived == false
+      - root_stats.certificates[0].dns_names == ["root.ansible.com"]
+      - root_stats.certificates[0].extensions | count == 3
+      - root_stats.certificates[0].has_private_key == false
+      - root_stats.certificates[0].issued_by == "root.ansible.com"
+      - root_stats.certificates[0].issued_to == "root.ansible.com"
+      - root_stats.certificates[0].issuer == "C=US, CN=root.ansible.com"
+      - root_stats.certificates[0].path_length_constraint == 0
+#      - root_stats.certificates[0].public_key == (root_pub.stdout_lines | join())
```

Comment:
Can this be cleaned up? If not, please add a comment on why it's needed. 
This applied to row 80 as well

Reply:
It was commented originally, so can't tell why. But I will remove it as it's an old code anyways and if it wasn't added back, it seems like it can be removed.